Sources/SwiftProtobuf/Message.swift

```diff
@@ -104,7 +107,8 @@ public protocol Message: CustomDebugStringConvertible {
 
   /// Overridden by well-known-types with custom JSON requirements.
   func serializeJSON() throws -> String
-  init(decoder: inout JSONDecoder) throws
+
+  mutating func decodeJSON(from: inout JSONDecoder) throws
```

Comment:
Note the parallel structure: `decodeJSON(from:)`, `decodeText(from:)`, and `decodeProtobuf(from:count:extensions:)`.

Hmmm...  I should review the `decodeProtobuf()` form and see if it can accept a `ProtobufDecoder` instead.